### ISSUE TITLE
FR (StorageManager): Option to upload file on click

### ISSUE THREAD

**Issue by ssierrac** [aws-amplify/amplify-ui#3796]
### On which framework/platform would you like to see this feature implemented?

React

### Which UI component is this feature-request for?

Storage (File Uploader)

### Please describe your feature-request in detail.

It would be nice to have the option to upload the file with one click and not just automatically when the file is selected

### We love contributors! Is this something you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request.
- [ ] ⚠️ This feature might incur a breaking change.

**Comment by dbanksdesign** [comment]
@SebastianSierrajc thank you for the request! I think this is a great idea

**Comment by dbanksdesign** [comment]
I know this is not the prettiest thing in the world, and having an option to upload files on click is still something we want to do, in the meantime there is a workaround... 

You can use the `processFile` function which can accept a Promise and make that promise wait until some external event or state happens. In this example I have it so the files don't start uploading until the submit button is clicked and then after all the files have uploaded it will show a message. 

https://stackblitz.com/edit/amplify-ui-storage-manager-kb66xx?file=src/App.tsx

![CleanShot 2023-05-08 at 22 34 37](https://user-images.githubusercontent.com/321279/237003920-763d011e-b185-4731-9bd6-b89f44c6554f.gif)

**Comment by Search-For-a-Friend** [comment]
Thank you for the workaround !
In my personal use case, I also need the user to be able to manage / cancel / confirm a batch of files for upload. So not starting uploading files as soon as dropped / selected is necessary. 
So I am interested in getting back this feature (which was present in FileUploader indeed).

**Comment by ioanabrooks** [comment]
@ssierrac, @Search-For-a-Friend We released this feature in version `2.3.0` of the `@aws-amplify/ui-react-storage` package, please give it a try and let us know! Please see docs for usage here: https://ui.docs.amplify.aws/react/connected-components/storage/storagemanager#manually-upload
